device/coordinate_manager.h

```diff
@@ -45,6 +46,17 @@ class CoordinateManager {
 
     CoordinateManager(CoordinateManager& other) = default;
```

Comment:
Since this is mostly translating CoreCoord_V1, and not providing additional functionality, maybe rename to CoreCoordTranslator, or something alike

Reply:
https://github.com/tenstorrent/tt-umd/issues/349